File: Class02/app/src/main/java/br/edu/ifsp/souza/charles/pdm/contador/MainActivity.java
package br.edu.ifsp.souza.charles.pdm.contador;

import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.Toast;

import br.edu.ifsp.souza.charles.pdm.contador.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    private ActivityMainBinding amb;
    private int contador = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        amb = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(amb.getRoot());

        amb.inicialSp.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
            @Override
            public void onItemSelected(AdapterView<?> adapterView, View view, int i, long l) {
                contador = (i == 0)? 0 : (i==1)? 5: 10;
            }

            @Override
            public void onNothingSelected(AdapterView<?> adapterView) {}
        });

        amb.cliqueBt.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                amb.contadorTv.setText(String.valueOf(++contador));
            }
        });
    }
}

File: Class09/app/src/main/java/br/edu/ifsp/souza/charles/pdm/mycontacts/view/MainActivity.kt
package br.edu.ifsp.souza.charles.pdm.mycontacts.view

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.widget.ArrayAdapter
import androidx.activity.result.ActivityResultLauncher
import androidx.activity.result.contract.ActivityResultContracts
import br.edu.ifsp.souza.charles.pdm.mycontacts.R
import br.edu.ifsp.souza.charles.pdm.mycontacts.databinding.ActivityMainBinding
import br.edu.ifsp.souza.charles.pdm.mycontacts.model.Constant.EXTRA_CONTACT
import br.edu.ifsp.souza.charles.pdm.mycontacts.model.Contact

class MainActivity : AppCompatActivity() {

    private val amb: ActivityMainBinding by lazy {
        ActivityMainBinding.inflate(layoutInflater)
    }
    // Data source
    private val contactList: MutableList<Contact> = mutableListOf()

    //Adaptar
    private val contactAdpater: ArrayAdapter<String> by lazy {
        ArrayAdapter(
            this,
            android.R.layout.simple_list_item_1,
            contactList.map{ it.toString()}
        )
    }

    private lateinit var  carl: ActivityResultLauncher<Intent>

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(amb.root)

        fillContactList()
        amb.contactsLv.adapter = contactAdpater

        carl = registerForActivityResult(
            ActivityResultContracts.StartActivityForResult(),
        ) { result ->
            if(result.resultCode == RESULT_OK){
              val contact = result.data?.getParcelableExtra<Contact>(EXTRA_CONTACT)
              contact?.let {  _contact ->
                  contactList.add(_contact)
                  contactAdpater.add(_contact.toString())
                  contactAdpater.notifyDataSetChanged()
              }
            }
        }
    }


    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        menuInflater.inflate(R.menu.menu_main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return when(item.itemId){
            R.id.addContactMi -> {
                carl.launch(Intent(this, ContactActivity::class.java))
                // Abre a tela de contato
                true
            }
            else -> {false}
        }
    }



    private fun fillContactList(){
        for ( i in 1 .. 10) {
            contactList.add(
                Contact(
                    id = i,
                    name = "Nome $i",
                    address = "Endereço $i",
                    phone = "Telefone $i",
                    email = "Email $i",
                )
            )
        }
    }
}